### PR TITLE
dictionary content provider

### PR DESCRIPTION
**Pull request by reyrahadian**
dictionary content provider

### CODE REVIEW

File: JabbR/ContentProviders/DictionaryContentProvider.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                pageInfo.WordDefinition = GetWordDefinition(htmlDocument);
+            }
+
+            return pageInfo;
+        }
+
+        private string GetWordDefinition(HtmlDocument htmlDocument)
+        {
+            var wordDefinition = htmlDocument.DocumentNode.SelectSingleNode("//div[@class=\"body\"]");
+            if (wordDefinition == null)
```
Comment (davidfowl) [inline JabbR/ContentProviders/DictionaryContentProvider.cs:56]: self closing if here.

File: JabbR/ContentProviders/DictionaryContentProvider.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                return string.Empty;
```
Comment (davidfowl) [inline JabbR/ContentProviders/DictionaryContentProvider.cs]: String.Empty

File: JabbR/ContentProviders/DictionaryContentProvider.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+            foreach (var link in links)
+            {
+                var href = link.Attributes["href"];
+                if (href != null && href.Value.StartsWith("/"))
+                {
+                    href.Value = string.Format("{0}{1}", _domain, href.Value);
+
+                    if (link.Attributes["style"] != null)
+                        link.Attributes["style"].Value = string.Empty;
```
Comment (davidfowl) [inline JabbR/ContentProviders/DictionaryContentProvider.cs]: Same as above.

File: JabbR/ContentProviders/DictionaryContentProvider.cs (hunk 1 above)
Comment (reyrahadian) [inline JabbR/ContentProviders/DictionaryContentProvider.cs:56]: fixed

File: JabbR/ContentProviders/DictionaryContentProvider.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+            foreach (var link in links)
+            {
+                var href = link.Attributes["href"];
+                if (href != null && href.Value.StartsWith("/"))
+                {
+                    href.Value = string.Format("{0}{1}", _domain, href.Value);
+
+                    if (link.Attributes["style"] != null)
+                        link.Attributes["style"].Value = String.Empty;
```
Comment (davidfowl) [inline JabbR/ContentProviders/DictionaryContentProvider.cs]: self closing if here too...

Comment (reyrahadian) [inline JabbR/ContentProviders/DictionaryContentProvider.cs]: missed that, sorry :(

File: JabbR/ContentProviders/DictionaryContentProvider.cs [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+            // fix relative url
+            var links = wordDefinition.SelectNodes("//a");
+
+            foreach (var link in links)
+            {
+                var href = link.Attributes["href"];
+                if (href != null && href.Value.StartsWith("/"))
+                {
+                    href.Value = string.Format("{0}{1}", _domain, href.Value);
```
Comment (davidfowl) [inline JabbR/ContentProviders/DictionaryContentProvider.cs]: String.Format uppercase s

**Pull request merged**